### PR TITLE
Allow RWX, FS infra PVC

### PR DESCRIPTION
**Pull request by nunnatsa**
**What this PR does / why we need it**:
For RWX pvc request, don't limit the infra PVC to be Block, but also allow FS.

**Release note**:

```release-note
For RWX pvc request, don't limit the infra PVC to be Block, but also allow FS.
```

### CODE REVIEW

File: e2e/create-pvc_test.go [hunk 1]
```diff
@@ -173,6 +173,70 @@ var _ = Describe("CreatePVC", func() {
 		deletePod(tenantClient.CoreV1(), namespace, writerPod.Name)
 	})
 
+	It("should creates a RW-Many block pvc, with FS infra storage class, and attaches to pod", Label("pvcCreation", "RWX", "Block", "infra-FS"), func() {
+		pvcName := "test-pvc"
+		storageClassName := "infra-fs"
+		pvc := pvcSpec(pvcName, storageClassName, "10Mi")
+		volumeMode := k8sv1.PersistentVolumeBlock
+		pvc.Spec.VolumeMode = &volumeMode
```
Comment (awels) [inline e2e/create-pvc_test.go:181]: Nit: maybe use ptr.To here instead of a separate variable?

**Review by awels — COMMENTED** [review]
looks good to me, just one little nit.

File: e2e/create-pvc_test.go (hunk 1 above)
Comment (nunnatsa) [reply e2e/create-pvc_test.go]: Thanks @awels . good point - fixed

**Review by awels — APPROVED** [review]
/approve

**Comment by awels** [comment]
Looks like maybe you need a rebase? We made some changes to the RBAC in particular for volume snapshots.

**Pull request merged**